examples/barotropic_tide.jl

```diff
@@ -0,0 +1,285 @@
+# # Barotropic tide
+#
+# In this example, we simulate the evolution of a barotropic tide over a hill.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, CairoMakie"
+# ```
+
+using Oceananigans
+using Oceananigans.Units
+
+# ## Grid
+
+# We use an immersed boundary two-dimensional grid (in ``x``--``z``) that is is periodic in
+# the ``x``-direction. To construct an immersed boundary grid we first need to create what
+# we refer to as "underlying grid", which the grid that encompasses the immersed boundary.
+
+Nx, Nz = 200, 80
+
+H  = 2kilometers
+Lx = 2000kilometers
+
+underlying_grid = RectilinearGrid(size = (Nx, Nz),
+                                  x = (-Lx/2, Lx/2),
+                                  z = (-H, 0),
+                                  halo = (4, 4),
+                                  topology = (Periodic, Flat, Bounded))
+
+# Now we can create the non-trivial bathymetry as an immersed boundary. We use `GridFittedBottom` that
+# gets as input either *(i)* a two-dimensional function whose arguments are the grid's native horizontal
+# coordinates and it returns the ``z`` of the bottom, or *(ii)* a two-dimensional array with the values
+# of ``z`` at the bottom at the cell centers.
+#
+# In this example we'd like to have a small Gaussian hill at the center of the domain.
+
+# ```math
+# h(x) = -H + h_0 \exp(-x^2 / 2σ^2)
+# ```
+
+h₀ = 50 # m
```

Comment:
why so tiny?

Reply:
why not?

Reply:
Its hard to see in an animation and requires more vertical resolution. It's also less classical. A "seamount" is a large feature. This is just an isolated hill in an abyssal plain, which doesn't really occur in the ocean.

Reply:
well, I'm happy to change it... I wanted to see the internal wave beam

do you have a paper in mind? perhaps I can try to replicate that?

Reply:
I can also call the example "internal tide from an abyssal hill"